pkg/controller/util/cluster_util.go

```diff
@@ -129,3 +129,10 @@ var KubeconfigGetterForSecret = func(secret *apiv1.Secret) clientcmd.KubeconfigG
 		return clientcmd.Load(data)
 	}
 }
+
+// IsPrimaryCluster checks if the caller is working with objects for the
```

Comment:
Consider adding (moving) a TODO indicating the need to revisit this when cluster id becomes a thing.